modules/renter/hostdb/hostdb.go

```diff
@@ -38,18 +65,51 @@ type HostDB struct {
 	// scan.
 	scanPool chan *hostEntry
 
-	mu *sync.RWMutex
+	mu sync.RWMutex
+}
+
+// a hostContract includes the original contract made with a host, along with
+// the most recent revision.
+type hostContract struct {
+	ID              types.FileContractID
+	FileContract    types.FileContract
+	LastRevisionTxn types.Transaction
+
+	// revisions must happen in serial
+	mu sync.RWMutex // TODO: consider `json:"-"`
 }
 
 // New creates and starts up a hostdb. The hostdb that gets returned will not
 // have finished scanning the network or blockchain.
-func New() *HostDB {
+func New(cs modules.ConsensusSet, wallet modules.Wallet, tpool modules.TransactionPool, persistDir string) (*HostDB, error) {
+	if cs == nil {
+		return nil, errNilCS
+	}
+	if wallet == nil {
+		return nil, errNilWallet
+	}
+	if tpool == nil {
+		return nil, errNilTpool
+	}
+
 	hdb := &HostDB{
+		wallet: wallet,
+		tpool:  tpool,
+
+		contracts:   make(map[types.FileContractID]hostContract),
 		activeHosts: make(map[modules.NetAddress]*hostNode),
 		allHosts:    make(map[modules.NetAddress]*hostEntry),
 		scanPool:    make(chan *hostEntry, scanPoolSize),
 
-		mu: sync.New(modules.SafeMutexDelay, 1),
+		persistDir: persistDir,
+	}
+	_, err := rand.Read(hdb.entropy[:])
```

Comment:
I forget what the hostdb entropy is used for. Something about determinism with a random number generator?